packages/core-mobile/app/contexts/SwapContext/performSwap/performSwap.ts

```diff
@@ -51,22 +50,15 @@ export async function performSwap({
   assert(activeNetwork, 'Network Init Error: Wrong network')
   assert(!activeNetwork.isTestnet, 'Network Init Error: Wrong network')
 
-  const spenderAddress = await getParaswapSpender()
-
   const sourceTokenAddress = isSrcTokenNative ? ETHER_ADDRESS : srcTokenAddress
   const destinationTokenAddress = isDestTokenNative
     ? ETHER_ADDRESS
     : destTokenAddress
 
-  const buildOptions = undefined,
-    partnerAddress = undefined,
-    partner = 'Avalanche',
-    receiver = undefined,
-    permit = undefined,
-    deadline = undefined,
-    partnerFeeBps = undefined
+  const partner = 'Avalanche'
 
   let approveTxHash: string | undefined
+  let spenderAddress: string
```

Comment:
I think you can declare `spenderAddress` inside the `if (!isSrcTokenNative)` statement.